src/java.base/share/classes/java/lang/classfile/ClassHierarchyResolver.java

```diff
@@ -222,7 +222,8 @@ static ClassHierarchyResolver of(Collection<ClassDesc> interfaces,
      * @return the class hierarchy resolver
      */
     static ClassHierarchyResolver ofClassLoading(ClassLoader loader) {
-        requireNonNull(loader);
+        //null check here breaks WithSecurityManagerTest
+//        requireNonNull(loader);
```

Comment:
The `null` `ClassLoader` is valid and refers to either the system or bootstrap class loader (depending on the API).

In the case of [`Class​::forName(…)`], it’s the bootstrap class loader (which is represented by `null` (`Object.class.getClassLoader()` returns `null`!))
```suggestion
```

[`Class​::forName(…)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Class.html#forName(java.lang.String,boolean,java.lang.ClassLoader)

Reply:
We intend to avoid using `null` to represent bootstrap class loader here, like in `MethodType`: https://github.com/openjdk/jdk/blob/bbb516163d400a9c7e923e423fe2a60091b59322/src/java.base/share/classes/java/lang/invoke/MethodType.java#L1197